src/libraries/System.Security.Cryptography/tests/X509Certificates/ChainPolicyTests.cs

```diff
@@ -0,0 +1,143 @@
+                source.VerificationTime = DateTime.MinValue;
+                source.VerificationTimeIgnored = false;
+                source.UrlRetrievalTimeout = TimeSpan.MaxValue;
+                source.VerificationFlags = X509VerificationFlags.IgnoreCtlNotTimeValid;
+                source.RevocationMode = X509RevocationMode.Offline;
+                source.RevocationFlag = X509RevocationFlag.EntireChain;
+
+                X509ChainPolicy clone = source.Clone();
+                Assert.Equal(source.VerificationTime, clone.VerificationTime);
+                Assert.Equal(source.VerificationTimeIgnored, clone.VerificationTimeIgnored);
+                Assert.Equal(source.VerificationFlags, clone.VerificationFlags);
+                Assert.Equal(source.RevocationFlag, clone.RevocationFlag);
+                Assert.Equal(source.RevocationMode, clone.RevocationMode);
+                Assert.Equal(source.UrlRetrievalTimeout, clone.UrlRetrievalTimeout);
+                Assert.Equal(source.DisableCertificateDownloads, clone.DisableCertificateDownloads);
+
+                Assert.NotSame(source.CertificatePolicy, clone.CertificatePolicy);
+                Assert.Equal(source.CertificatePolicy, clone.CertificatePolicy);
+                Assert.Same(source.CertificatePolicy[0], clone.CertificatePolicy[0]);
+
+                Assert.NotSame(source.ApplicationPolicy, clone.ApplicationPolicy);
+                Assert.Equal(source.ApplicationPolicy, clone.ApplicationPolicy);
+                Assert.Same(source.ApplicationPolicy[0], clone.ApplicationPolicy[0]);
+
+                Assert.NotSame(source.ExtraStore, clone.ExtraStore);
+                Assert.Equal(source.ExtraStore, clone.ExtraStore);
+                Assert.Same(source.ExtraStore[0], clone.ExtraStore[0]);
+
+                Assert.NotSame(source.CustomTrustStore, clone.CustomTrustStore);
+                Assert.Equal(source.CustomTrustStore, clone.CustomTrustStore);
+                Assert.Same(source.CustomTrustStore[0], clone.CustomTrustStore[0]);
+
+                Assert.NotSame(source.ExtraStore[0], clone.CustomTrustStore[0]);
+            }
+        }
+
+        private static void AssertDefaultState(X509ChainPolicy policy)
+        {
+            Assert.Equal(X509RevocationFlag.ExcludeRoot, policy.RevocationFlag);
+            Assert.Equal(X509RevocationMode.Online, policy.RevocationMode);
+            Assert.Equal(X509VerificationFlags.NoFlag, policy.VerificationFlags);
+            Assert.Equal(X509ChainTrustMode.System, policy.TrustMode);
+            Assert.False(policy.DisableCertificateDownloads, "policy.DisableCertificateDownloads");
+            Assert.True(policy.VerificationTimeIgnored, "policy.VerificationTimeIgnored");
+            Assert.Equal(TimeSpan.Zero, policy.UrlRetrievalTimeout);
+
+            // A fairly generous tolerance of an NTP adjustment.
+            Assert.Equal(DateTime.Now, policy.VerificationTime, TimeSpan.FromMinutes(30));
```

Comment:
We've run in to DST changes issues before, e.g. https://github.com/dotnet/runtime/issues/61302

Since we are dealign with `DateTime.Now`, not `UtcNow`, is there a chance this could fail when there's a DST change?